src/platform/SDL2Window.cxx

```diff
@@ -165,6 +171,24 @@ void SDLWindow::swapBuffers()
 #endif //__APPLE__
 }
 
+// For some reason, when creating a new fullscreen window on Linux with a different resolution than before, SDL throws
+// a resize event with the old window resolution, which is not what we want. This function is called to filter SDL
+// window resize events right after the resolution change and adjust the resolution to the correct one.
+#ifdef __linux__
```

Comment:
does it **need** to be restricted to Linux?

Reply:
Not sure. In general, I would like to keep platform-specific workarounds within `#ifdef` guards, because we have a tendency to break other platforms while fixing an issue on one of them. Keeping the unknowns to a minimum is a plus in my opinion.